R/render-table.R

```diff
@@ -37,23 +104,96 @@ renderTable <- function(expr, ..., env=parent.frame(), quoted=FALSE, func=NULL)
     xtable_args <- dots[intersect(names(dots), xtable_argnames)]
     non_xtable_args <- dots[setdiff(names(dots), xtable_argnames)]
 
-    # Call xtable with its args
+    # A small helper function to determine if the row.names can be coerced
+    # to numeric or if they are legitimate strings
+    isNumber <- function(n) {
+      grepl("^\\d$+", n)
```

Comment:
Two things:

* I think it should be `grepl("^\\d+$", n)`. Right now:

```R
isNumber("12")
# FALSE
```
 
* For the use of `isNumber` below, you need `all()` somewhere. For example:

```R
if (isNumber(c("1", "A"))) print("number")
# [1] "number"
# Warning message:
# In if (isNumber(c("1", "A"))) print("number") :
#   the condition has length > 1 and only the first element will be used
```

So I'd suggest renaming this to `allNumbers` and use `all(grepl(....))` inside.